src/Controller/AdminCourseController.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace App\Controller;
+
+class AdminCourseController extends AbstractController
+{
+    private const INPUTS_VALIDATIONS = [
+        'name' => 255,
+        'day' => 8,
+        'time' => '',
+        'duration' => 120,
+        'teacher' => '',
+        'capacity' => 20,
+    ];
+
+    public function course(): string
+    {
+        $errors = [];
+        $formData = [];
+        if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+            $formData = array_map('trim', $_POST);
+            $errors = $this->validateFormulary($formData);
+            if (empty($errors)) {
+                header('location: /admin/course');
+            }
+        }
+
+        return $this->twig->render('Admin/course.html.twig', [
+            'formulary' => $formData,
+            'errors' => $errors,
+        ]);
+    }
+
+    /**
+     * Test fields if exist, if not empty and test their constraints
+     *
+     * @param array $formData
+     * @return array
+     */
+    private function validateFormulary(array $formData): array
+    {
+
+        $errors = [];
+
+        foreach (array_keys($formData) as $inputName) {
+            if (!array_key_exists($inputName, self::INPUTS_VALIDATIONS)) {
+                $errors[] = 'Le champs ' . $inputName . ' n\'existe pas.';
+            }
+        }
```

Comment:
To test if the field exists. If it has not been modified by the user. 
Is it too much at the moment ? Because if the field is changed, it breaks all the tests. 
If ok for you I have to see how to handle it. if a field name is changed I stop everything and return only these errors?